test/jdk/java/awt/dnd/InterJVMGetDropSuccessTest/InterJVMGetDropSuccessTest.java

```diff
@@ -265,46 +265,49 @@ public void run(String[] args) {
             Point targetPoint = new Point(x + w / 2, y + h / 2);
 
             Robot robot = new Robot();
+            robot.setAutoWaitForIdle(true);
             robot.mouseMove(sourcePoint.x, sourcePoint.y);
-            robot.mousePress(InputEvent.BUTTON1_MASK);
+            Thread.sleep(50);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
             for (Point p = new Point(sourcePoint); !p.equals(targetPoint);
                  p.translate(Util.sign(targetPoint.x - p.x),
                              Util.sign(targetPoint.y - p.y))) {
                 robot.mouseMove(p.x, p.y);
                 Thread.sleep(50);
             }
 
+            boolean success1 = false;
             synchronized (Util.SYNC_LOCK) {
-                robot.mouseRelease(InputEvent.BUTTON1_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
                 Util.SYNC_LOCK.wait(Util.FRAME_ACTIVATION_TIMEOUT);
+                if (!dragSourceListener.isDropFinished()) {
+                    throw new RuntimeException("Drop not finished");
+                }
+                success1 = dragSourceListener.getDropSuccess();
+                dragSourceListener.reset();
             }
 
-            if (!dragSourceListener.isDropFinished()) {
-                throw new RuntimeException("Drop not finished");
-            }
-
-            boolean success1 = dragSourceListener.getDropSuccess();
 
-            dragSourceListener.reset();
             robot.mouseMove(sourcePoint.x, sourcePoint.y);
-            robot.mousePress(InputEvent.BUTTON1_MASK);
+            Thread.sleep(50);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
             for (Point p = new Point(sourcePoint); !p.equals(targetPoint);
                  p.translate(Util.sign(targetPoint.x - p.x),
                              Util.sign(targetPoint.y - p.y))) {
                 robot.mouseMove(p.x, p.y);
                 Thread.sleep(50);
             }
 
+            boolean success2 = false;
             synchronized (Util.SYNC_LOCK) {
-                robot.mouseRelease(InputEvent.BUTTON1_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
```

Comment:
I suppose this would work but the code is hard to read.
1. You are synchronizing on SYNC_LOCK on multiple places: sometimes in the listener method and sometimes around the listener methods.
2. The SYNC_LOCK was originally needed just to call wait/notify, you are using it for more than that now, which is less transparent.
3. Finally the access to the listener's field is already partially guarded as method calls from the drag happen on the AWT event queue. It would be easier to understand if calls to the methods which originates from main would also go through the event queue. That is, after all, the most common approach for the java UI code.